x/leveragelp/keeper/begin_blocker.go

```diff
@@ -15,47 +16,41 @@ func (k Keeper) BeginBlocker(ctx sdk.Context) {
 	// check if epoch has passed then execute
 	epochLength := k.GetEpochLength(ctx)
 	epochPosition := k.GetEpochPosition(ctx, epochLength)
+	params := k.GetParams(ctx)
 
-	if epochPosition == 0 { // if epoch has passed
-		pools := k.GetAllPools(ctx)
+	if epochPosition == 0 && params.FallbackEnabled { // if epoch has passed
+		pageReq := &query.PageRequest{
+			Limit:      uint64(params.NumberPerBlock),
+			CountTotal: true,
+		}
+		offset, _ := k.GetOffset(ctx)
+		pageReq.Offset = offset
+		positions, _, err := k.GetPositions(ctx, pageReq)
+		if err != nil {
+			ctx.Logger().Error(errors.Wrap(err, fmt.Sprintf("error fetching paginated positions")).Error())
+			return
```

Comment:
We should have panic here. Any error that shouldn't happen should directly stop the chain (though it might not happen even with panic)

Reply:
why ? I think this is fine, we will get the log, panic will stop the chain I guess

Reply:
Tendermint has panic handlers and doesn't allow the chain to stop unless there is consensus failure (if you get panic inside a tx, the chain doesn't stop but has a whole stack trace in the tx log). The logger here won't print the stack trace, the return undermines the severity of the issue and overall it's better to stop the chain and fix it if there is a critical error that might lead to users losing money.

Reply:
1. That's not case here, this is not tx, it is `begin_block`, I have attached a ss for your reference, you can try it on your system as well, (put panic in begin blocker)
![Screenshot 2024-08-09 at 12 35 50 PM](https://github.com/user-attachments/assets/0a4911ec-7804-4cbc-92d7-bfe4685b3177)

2. We have the required log in error message, so we will know this is the point of failure

3. About severity of issue, so this is a fallback system and we won't see error while fetching pages but if that occurs, we need to handle the error, that's why this log is added so we know the source of error

4. We will fix it and it won't panic the chain (that means switching all of the chain resulting in more funds loss)

Reply:
I'll trust your judgment on this matter. In the case of certain chains (such as Cosmos or Osmosis), the BeginBlock function does include a call to panic(err) when necessary. This approach is typically used because the error should always be handled appropriately, and if it cannot be returned correctly, the chain is designed to trigger a panic.

Reference: https://github.com/cosmos/cosmos-sdk/blob/v0.47.11/x/distribution/keeper/allocation.go#L25
The AllocateTokens is called in the BeginBlock of the distribution module. It distributes block rewards to validators

Reply:
for modules such as bank, distribution etc, these are core modules and necessary to panic if a user don't rewards, question is do we want to halt the chain (stop swaps, perps, etc)because of fallback system not getting paginated queries to liquidate positions ?

Reply:
okay, so I checked osmosis codebase too, here we have
 https://github.com/osmosis-labs/osmosis/blob/977929f2a4a644318e78b51b510f595e581802e8/x/downtime-detector/abci.go#L11
and 
https://github.com/osmosis-labs/osmosis/blob/977929f2a4a644318e78b51b510f595e581802e8/x/superfluid/abci.go#L11 panic here
and at other places we don't panic, looking at out use case I think we should not panic here